cmake/CPM.cmake

```diff
@@ -28,7 +28,7 @@
 
 cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
 
-set(CURRENT_CPM_VERSION 0.33.0)
+set(CURRENT_CPM_VERSION 0.34.0)
```

Comment:
I don't wanna be nit-picky but here a single commit is doing multiple things. In the long run, I prefer changes to be isolated from each other (one commit for CPM version upgrade, another commit for e.g. a feature addition, another commit for e.g. cmake options changes)

Reply:
I'll update it in other PR. :)